classes/local/manage_rule_plugins.php

```diff
@@ -0,0 +1,80 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_registrationrules\local;
```

Comment:
I've not thought about it before, but as it's part of the admin settings it is able to reside in `tool_registrationrules\admin` namespace according to level 2 namespace rules.

What are your thoughts about placing it?
I usually try to avoid local if there's an alternative.

Reply:
I see no harm in it, seems to make sense semantically too :)